endpoints/openrtb2/auction_test.go

```diff
@@ -1126,31 +1136,104 @@ func TestDisabledBidder(t *testing.T) {
 	}
 }
 
-func TestValidateImpExtDisabledBidder(t *testing.T) {
-	imp := &openrtb.Imp{
-		Ext: json.RawMessage(`{"appnexus":{"placement_id":555},"unknownbidder":{"foo":"bar"}}`),
+func TestValidateImpExt(t *testing.T) {
+	testCases := []struct {
+		description  string
+		requestExt   json.RawMessage
+		expectedExt  string
+		expectedErrs []error
+	}{
+		{
+			description:  "Empty",
+			requestExt:   nil,
+			expectedExt:  "",
+			expectedErrs: []error{errors.New("request.imp[0].ext is required")},
+		},
+		{
+			description:  "Valid Bidder",
+			requestExt:   json.RawMessage(`{"appnexus":{"placement_id":555}}`),
+			expectedExt:  `{"appnexus":{"placement_id":555}}`,
+			expectedErrs: []error{},
+		},
+		{
+			description:  "Valid Bidder + Disabled Bidder",
+			requestExt:   json.RawMessage(`{"appnexus":{"placement_id":555},"unknownbidder":{"foo":"bar"}}`),
+			expectedExt:  `{"appnexus":{"placement_id":555}}`,
+			expectedErrs: []error{&errortypes.BidderTemporarilyDisabled{Message: "The bidder 'unknownbidder' has been disabled."}},
+		},
+		{
+			description:  "Valid Bidder + Disabled Bidder + First Party Data Context",
+			requestExt:   json.RawMessage(`{"appnexus":{"placement_id":555},"unknownbidder":{"foo":"bar"},"context":{"data":{"keywords":"prebid server example"}}}`),
+			expectedExt:  `{"appnexus":{"placement_id":555},"context":{"data":{"keywords":"prebid server example"}}}`,
+			expectedErrs: []error{&errortypes.BidderTemporarilyDisabled{Message: "The bidder 'unknownbidder' has been disabled."}},
+		},
+		{
+			description:  "Valid Bidder + First Party Data Context",
+			requestExt:   json.RawMessage(`{"appnexus":{"placement_id":555},"context":{"data":{"keywords":"prebid server example"}}}`),
+			expectedExt:  `{"appnexus":{"placement_id":555},"context":{"data":{"keywords":"prebid server example"}}}`,
+			expectedErrs: []error{},
+		},
+		{
+			description:  "Valid Prebid Ext Bidder",
+			requestExt:   json.RawMessage(`{"prebid":{"bidder":{"appnexus":{"placement_id":555}}}}`),
+			expectedExt:  `{"prebid":{"bidder":{"appnexus":{"placement_id":555}}}}`,
+			expectedErrs: []error{},
+			// request.ext.prebid.bidder.{biddername} is only promoted/copied to request.ext.{biddername} if there is at least one disabled bidder.
```

Comment:
We need to talk about this behavior. It makes no sense to me and looks like a bug. 
- I can understand removing a disabled bidder, but that's not happening for imp.ext.bidder definitions.
- I have no idea why we're copying the bidder up a level here. That hoisting is already performed by the exchange during imp sanitization. Maybe this was added first and never removed?

Reply:
I thought about this more and now consider it a bug. As a follow-up PR, I'd like to make the validate method only actually validate and move the disabled bidder logic to `exchange.go`, if it's not already there.